packages/mui-base/src/Button/Button.tsx

```diff
@@ -167,4 +167,6 @@ Button.propTypes /* remove-proptypes */ = {
   to: PropTypes.string,
 } as any;
 
+Button.displayName = 'Button';
```

Comment:
What about?

```suggestion
if(process.env.NODE_ENV !== 'production') {
  Button.displayName = 'Button';
}
```

this would mean:

1. be sure there is no regressions
2. give developers developer confidence that this isn't going to bloat their bundle
3. In the code ejection story, we could automatically wrap this, but I would expect people to also copy and paste the source too

Reply:
I thought about it but went for the current solution because

- Parity with the proptypes generation. Both work the exact same way: upsert code => wrap in build step. If you understand one, you also understand the other. If we add it in for displayName, then I think we should do the same for `propTypes` and remove the babel plugin.
- It's more straightforward for the detection logic if it consists of a single top-level statement statement
- It prevents people from getting tempted adding more things in the coditional. Which in turn would either break the detection logic, or make it much more complex.

Reply:
> Parity with the proptypes generation

Ah right ok, I forgot, I thought we were doing the opposite. Makes sense then 👍